### ISSUE TITLE
Crash when required arguments are not given / passed.

### ISSUE THREAD

**Issue by LipkeGu** [OpenRA/OpenRA#8442]
it is able to crash the Utitlity by simpliy omitting required arguments :/

```
Received args: --map-import
System.IndexOutOfRangeException: Der Index war außerhalb des Arraybereichs.
   bei OpenRA.Mods.Common.UtilityCommands.ImportLegacyMapCommand.Run(ModData modData, String[] args)
   bei OpenRA.Utility.Program.Main(String[] args) in d:\Projekte\OpenRA\OpenRA.Utility\Program.cs:Zeile 54.

```

**Comment by Phrohdoh** [comment]
This is a completely normal part of applications that require arguments.

Unless you are suggest we just give a warning and catch any exceptions (or implement nargs).

**Comment by LipkeGu** [comment]
we should redirect the user to the Usage function in this case

**Comment by penev92** [comment]
This is a completely normal part of applications that require arguments.

And also it is unreasonable to add argument checks for every utility
command (they will each need a separate check).
On Jun 8, 2015 3:10 PM, "Taryn Hill" <notifications@github.com> wrote:

This is a completely normal part of applications that require arguments.

Unless you are suggest we just give a warning and catch any exceptions (or
implement nargs).

—
Reply to this email directly or view it on GitHub
<https://github.com/OpenRA/OpenRA/issues/8442#issuecomment-109970222>.

**Comment by Phrohdoh** [comment]
@penev92 Well, as I was saying, we could add `int nargs = -1` (default to 'any'), and check the number of arguments based on the `IUtilityCommand`'s `ArgumentCount` field.

**Comment by pchote** [comment]
> And also it is unreasonable to add argument checks for every utility command (they will each need a separate check).

I disagree.  It is completely reasonable to expect each command to check that it has been given proper input.

**Comment by obrakmann** [comment]
Fixed by #9397
